### PR TITLE
Have Mount use AHRS singleton, use altitude frame

### PR DESCRIPTION
**Pull request by peterbarker**
New tests added.

Tested by flashing onto my Solo and testing by hand.

### CODE REVIEW

File: libraries/AP_Mount/AP_Mount_Backend.cpp [hunk 1]
```diff
@@ -134,9 +134,13 @@ float AP_Mount_Backend::angle_input_rad(const RC_Channel* rc, int16_t angle_min,
 // calc_angle_to_location - calculates the earth-frame roll, tilt and pan angles (and radians) to point at the given target
 void AP_Mount_Backend::calc_angle_to_location(const struct Location &target, Vector3f& angles_to_target_rad, bool calc_tilt, bool calc_pan, bool relative_pan)
 {
-    float GPS_vector_x = (target.lng-_frontend._current_loc.lng)*cosf(ToRad((_frontend._current_loc.lat+target.lat)*0.00000005f))*0.01113195f;
-    float GPS_vector_y = (target.lat-_frontend._current_loc.lat)*0.01113195f;
-    float GPS_vector_z = (target.alt-_frontend._current_loc.alt);                 // baro altitude(IN CM) should be adjusted to known home elevation before take off (Set altimeter).
+    Location current_loc;
+    if (!AP::ahrs().get_position(current_loc)) {
```
Comment (rmackay9) [inline libraries/AP_Mount/AP_Mount_Backend.cpp]: I'm surprised this is the right thing to do actually.  I suspect it would be scope creep but really it should probably return a false and then the caller should handle that situation.

File: libraries/AP_Mount/AP_Mount_Backend.cpp [hunk 2]
```diff
@@ -131,16 +132,35 @@ float AP_Mount_Backend::angle_input_rad(const RC_Channel* rc, int16_t angle_min,
...
+}
+
 // calc_angle_to_location - calculates the earth-frame roll, tilt and pan angles (and radians) to point at the given target
-void AP_Mount_Backend::calc_angle_to_location(const struct Location &target, Vector3f& angles_to_target_rad, bool calc_tilt, bool calc_pan, bool relative_pan)
+bool AP_Mount_Backend::calc_angle_to_location(const struct Location &target, Vector3f& angles_to_target_rad, bool calc_tilt, bool calc_pan, bool relative_pan)
 {
     Location current_loc;
     if (!AP::ahrs().get_position(current_loc)) {
-        // completely ignore this error; ahrs will give us its best guess
+        return false;
```
Comment (rmackay9) [inline libraries/AP_Mount/AP_Mount_Backend.cpp]: Ah, you've fixed it!  ignore/resolve my previous comment.

File: libraries/AP_Mount/AP_Mount_Backend.cpp [hunk 3]
```diff
@@ -20,6 +20,7 @@ void AP_Mount_Backend::set_roi_target(const struct Location &target_loc)
 {
     // set the target gps location
     _state._roi_target = target_loc;
+    _state._roi_target_set_ms = AP_HAL::millis();
```
Comment (rmackay9) [inline libraries/AP_Mount/AP_Mount_Backend.cpp]: Do we ever use this time?  If not I guess it might as well be a boolean..

**Comment by peterbarker** [comment]
On Sun, 3 Nov 2019, Randy Mackay wrote:

> +    _state._roi_target_set_ms = AP_HAL::millis();
> 
> Do we ever use this time? If not I guess it might as well be a boolean..

I thought it might be useful to timeout on at some stage.  But yeah, I'll 
change it.

**Comment by peterbarker** [comment]
Tested this using MAVProxy.

Hacked up the gimbal driver to use the mount message rather than the gimbal report message, and master acts the same as this.

The cameraview module isn't set up to take the gimbal/mount messages at all, just servo outputs and stabilisation, so it's not suitable for testing.
